src/Core/Error.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @param {Element} element
+ * @param {*} err
+ * @param {string} from
+ * @param {number} signature
+ * @param {Element}
+ */
+function invokeErrorBoundary (element, err, from, signature) {
+	return commitElement(getErrorElement(element, getErrorException(element, err, from), from, signature))
+}
+
+/**
+ * @param {Element} element
+ * @param {Error} error
+ * @param {string} from
+ * @param {number} signature
+ * @return {Element?}
+ */
+function getErrorElement (element, error, from, signature) {
+	if (signature === SharedErrorPassive || !isValidElement(element) || !element.id === SharedElementEmpty)
+		return reportErrorException(error)
+
+	var boundary = element.owner && !!element.owner[SharedComponentDidCatch] 
+	var host = element.host
+	var time = element.time
+
+	requestAnimationFrame(function () {
+		if (element.active)
+			recoverErrorBoundary(element, commitElement(null))
+	})
+
+	if (boundary)
+		if (boundary = (element.time = Date.now()) - time > 16) {
+			element.work = SharedWorkUpdating
+			getLifecycleBoundary(element, SharedComponentDidCatch, error, error)
+			element.work = SharedWorkIdle
+		}
+
+	return getErrorElement(!boundary && host, error, from, signature)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ */
+function recoverErrorBoundary (element, snapshot) {
+	reconcileElement(getElementChildren(element), commitElement(snapshot))
+}
+
+/**
+ * @param {Error} error
+ */
+function reportErrorException (error) {
+	if (error.defaultPrevented)
+		return
+
+	console.error(error.stack + '\n\n' + error.errorMessage)
+}
+
+/**
+ * @param {*} value
+ * @return {Object}
+ */
+function getErrorDescription (value) {
+	return {enumerable: true, configurable: true, value: value}
```

Comment:
Consider making these properties non-enumerable, providing equivalent `util.inspect` (node.js console.log) output with errors

e.g.
```js
console.log(new Error('x'))
/*
Error x:
  at ...
  at ...
*/
console.log(dioError)
/*
{ Error x:
  at ...
  at ...
  
  errorLocation: 'render',
  errorMessage: 'xxxx',
  componentStack: ' xxxxxx',
  defaultPrevented: false,
  preventDefault: [Function] }
*/
```

Alternatively provide a [`util.inspect.custom` definition](https://nodejs.org/api/util.html#util_custom_inspection_functions_on_objects)

Reply:
The reason they are enumerable was to make destructuring work as expected. I'll explore the `inspect` option

Reply:
Implemented a `inspect` interface in the recent commit, though i couldn't get the `options.stylize('...', 'red')` to actually work and show color or maybe it's just my terminal settings?